From a1a93b3e6d5428228f00c227f327ae43f8541511 Mon Sep 17 00:00:00 2001
From: nixpkgs <nix@nixpkgs>
Date: Wed, 1 Apr 2026 17:56:21 +0300
Subject: [PATCH 3/3] remove Liquid Glass (macOS 26 API) for SDK 14
 compatibility


diff --git a/src/logic/Appearance.swift b/src/logic/Appearance.swift
index fdec38e..5688c63 100644
--- a/src/logic/Appearance.swift
+++ b/src/logic/Appearance.swift
@@ -85,12 +85,7 @@ class Appearance {
         } else {
             lightTheme()
         }
-        // for Liquid Glass, we don't want a shadow around the panel
-        if #available(macOS 26.0, *), currentStyle == .appIcons && LiquidGlassEffectView.canUsePrivateLiquidGlassLook() {
-            enablePanelShadow = false
-        } else {
-            enablePanelShadow = true
-        }
+        enablePanelShadow = true
     }
 
     private static func thumbnailsSize(_ isHorizontalScreen: Bool, _ size: AppearanceSizePreference) {
diff --git a/src/ui/main-window/TilesPanelBackgroundView.swift b/src/ui/main-window/TilesPanelBackgroundView.swift
index badf899..8ec6f90 100644
--- a/src/ui/main-window/TilesPanelBackgroundView.swift
+++ b/src/ui/main-window/TilesPanelBackgroundView.swift
@@ -1,39 +1,4 @@
-@available(macOS 26.0, *)
-class LiquidGlassEffectView: NSGlassEffectView, EffectView {
-    private typealias SetVariantType = @convention(c) (AnyObject, Selector, Int) -> Void
-    private static let setVariantSelector = NSSelectorFromString("set_variant:")
 
-    static func canUsePrivateLiquidGlassLook() -> Bool {
-        let method = class_getInstanceMethod(object_getClass(NSGlassEffectView()), setVariantSelector)
-        return method != nil
-    }
-
-    convenience init(_ clear: Bool) {
-        self.init()
-        if clear {
-            style = .clear
-            safeSetVariant(3)
-        } else {
-            style = .regular
-        }
-        updateAppearance()
-        wantsLayer = true
-        // without this, there are weird shadows around the corners
-        layer!.masksToBounds = true
-    }
-
-    func safeSetVariant(_ value: Int) {
-        if let method = class_getInstanceMethod(object_getClass(self), LiquidGlassEffectView.setVariantSelector) {
-            let methodImplementation = method_getImplementation(method)
-            let f = unsafeBitCast(methodImplementation, to: SetVariantType.self)
-            f(self, LiquidGlassEffectView.setVariantSelector, value)
-        }
-    }
-
-    func updateAppearance() {
-        cornerRadius = Appearance.windowCornerRadius
-    }
-}
 
 class FrostedGlassEffectView: NSVisualEffectView, EffectView {
     convenience init(_: Int?) {
@@ -74,21 +39,5 @@ protocol EffectView: NSView {
 }
 
 func makeAppropriateEffectView() -> EffectView {
-    if #available(macOS 26.0, *) {
-        if Preferences.appearanceStyle == .appIcons {
-            if LiquidGlassEffectView.canUsePrivateLiquidGlassLook() {
-                Logger.debug { "Using LiquidGlassEffectView(true)" }
-                return LiquidGlassEffectView(true)
-            }
-            Logger.error {
-                let os = ProcessInfo.processInfo.operatingSystemVersion
-                let version = "\(os.majorVersion).\(os.minorVersion).\(os.patchVersion)"
-                return "Private API set_variant is no longer available. macOS version: \(version))"
-            }
-        }
-        Logger.debug { "Using LiquidGlassEffectView(false)" }
-        return LiquidGlassEffectView(false)
-    }
-    Logger.debug { "Using FrostedGlassEffectView(nil)" }
     return FrostedGlassEffectView(nil)
 }
-- 
2.53.0

